salt/modules/status.py

```diff
@@ -544,12 +544,20 @@ def version():
 
         salt '*' status.version
     '''
-    procf = '/proc/version'
-    if not os.path.isfile(procf):
-        return {}
-    ret = salt.utils.fopen(procf, 'r').read().strip()
-
-    return ret
+    def linux_version():
+        procf = '/proc/version'
+        if not os.path.isfile(procf):
+            return {}
+        return salt.utils.fopen(procf, 'r').read().strip()
+
+    # dict that return a function that does the right thing per platform
+    get_version = {
+    'Linux': linux_version,
+    'FreeBSD': lambda : __salt__['cmd.run']('sysctl -n kern.version'),
+    }
+
+    errmsg = 'This method is unsupported on the current operating system!'
+    return get_version.get(__grains__['kernel'], lambda: errmsg)()
```

Comment:
If we can't generate the grain we don't want to send an error, an empty grain value would be better, or no grain at all, also, this will cause the grain loader to stack trace as the error will not be a dict, so please just omit the error for an empty dict :)